docs/data/data-grid/row-grouping/row-grouping.md

```diff
@@ -97,39 +109,37 @@ const ratingColDef: GridColDef = {
 
 {{"demo": "RowGroupingCustomCellDefault.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### Show values for the leaves
+### Show values for leaves
 
-By default, the grouped rows display no value on their grouping columns' cells. Those cells are called "leaves."
+By default, grouped rows display no value in their grouping column cells—these cells are called **leaves**.
+To display a value in a leaf, provide the `leafField` property to the `groupingColDef`.
 
-If you want to display some value, you can provide a `leafField` property to the `groupingColDef`.
+The demo below uses `leafField` to display values from the **Title** column in the leaves.
 
 {{"demo": "RowGroupingLeafWithValue.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### Hide the descendant count
+### Hide descendant count
 
-Use the `hideDescendantCount` property of the `groupingColDef` to hide the number of descendants of a grouping row.
+By default, row group cells display the number of descendant rows they contain, in parentheses.
+Use the `hideDescendantCount` property of the `groupingColDef` to hide this number.
 
 {{"demo": "RowGroupingHideDescendantCount.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### Hide the grouped columns
-
-By default, the columns used to group the rows remains visible.
-For instance if you group by `"director"`, you have two columns titled **Director**:
-
-- The grouped column (the column from which you grouped the rows)
-- The grouping column on which you can toggle the groups
+### Hide grouped columns
 
-To automatically hide the grouped columns, use the `useKeepGroupedColumnsHidden` utility hook.
-The hook automatically hides the columns when added to the model, and displays them when removed from it.
+By default, [grouped columns](#grouping-columns) remain visible when grouping is applied.
+This means that when there's only one grouping criterion, the grouping column and the grouped column both display the same values, which may be redundant or unnecessary.
+You can use the `useKeepGroupedColumnsHidden` utility hook to hide grouped columns.
+This automatically hides grouped columns when added to the model, and displays them when removed.
 
 :::warning
 This hook is not compatible with the deprecated column property `hide`.
 
 You can manage column visibility with `columnVisibilityModel`, `initialState`, or both together.
-To do so, pass them to the hook parameters.
+To do so, pass them to the hook parameters as shown below.
 :::
```

Comment:
This property (`GridColDef['hide']`) is long [gone](https://github.com/mui/mui-x/pull/6505/files#diff-654c76d2458fcc7257077878eb1d07bbc743abae9c219ddb3c9296f1a50ffa36R61). 😄 
Not sure why it's still mentioned. I'd suggest to remove this callout completely.